### PR TITLE
CDAP-1495: Max table width for AsciiTable

### PR DESCRIPTION
**Pull request by jwang47**
```
    new AsciiTable<String>(
      new String[] {"c1", "c2", "c3333"},
      ImmutableList.of(
        "r1zz" + Strings.repeat("z", 300) + "456,r11,r1",
        "r2,r2222 zzzzzzz z z z zzzzzz z zzzzzzzzz zzzzzzz zzzzzzz zzzzzzz zzzzz zzz123,r",
        "r3333,r3,r3\n1"
      ),
      new RowMaker<String>() {
        @Override
        public Object[] makeRow(String object) {
          return object.split(",");
        }
      }
    ).print(System.out);
```

now outputs the following table:

```
+========================================================================================+
| c1                         | c2                         | c3333                        |
+========================================================================================+
| r1zzzzzzzzzzzzzzzzzzzzzzzz | r11                        | r1                           |
| zzzzzzzzzzzzzzzzzzzzzzzzzz |                            |                              |
| zzzzzzzzzzzzzzzzzzzzzzzzzz |                            |                              |
| zzzzzzzzzzzzzzzzzzzzzzzzzz |                            |                              |
| zzzzzzzzzzzzzzzzzzzzzzzzzz |                            |                              |
| zzzzzzzzzzzzzzzzzzzzzzzzzz |                            |                              |
| zzzzzzzzzzzzzzzzzzzzzzzzzz |                            |                              |
| zzzzzzzzzzzzzzzzzzzzzzzzzz |                            |                              |
| zzzzzzzzzzzzzzzzzzzzzzzzzz |                            |                              |
| zzzzzzzzzzzzzzzzzzzzzzzzzz |                            |                              |
| zzzzzzzzzzzzzzzzzzzzzzzzzz |                            |                              |
| zzzzzzzzzzzzzzzzzzzzzzzzzz |                            |                              |
|----------------------------------------------------------------------------------------|
| r2                         | r2222 zzzzzzz z z z zzzzzz | r                            |
|                            | z z zzzzzzzzz zzzzzzz zzzz |                              |
|----------------------------------------------------------------------------------------|
| r3333                      | r3                         | r3                           |
|                            |                            | 1                            |
+========================================================================================+
```

previously, AsciiTable would've output the following:

```
+===========================================================================================================================================================================================================================================================================================================================================================================================================+
| c1                                                                                                                                                                                                                                                                                                                  | c2                                                                          | c3333 |
+===========================================================================================================================================================================================================================================================================================================================================================================================================+
| r1zzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzz456 | r11                                                                         | r1    |
|-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| r2                                                                                                                                                                                                                                                                                                                  | r2222 zzzzzzz z z z zzzzzz z zzzzzzzzz zzzzzzz zzzzzzz zzzzzzz zzzzz zzz123 | r     |
|-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| r3333                                                                                                                                                                                                                                                                                                               | r3                                                                          | r3    |
|                                                                                                                                                                                                                                                                                                                     |                                                                             | 1     |
+===========================================================================================================================================================================================================================================================================================================================================================================================================+
```

### CODE REVIEW

**Comment by jwang47** [comment]
May have to make CLIMainTest more robust, as it simply checks for a substring in an output, but now that `AsciiTable` is splitting cell content if it exceeds the column width, some tests may start failing.

**Comment by jwang47** [comment]
Fix for `CLIMainTest` is in PR https://github.com/caskdata/cdap/pull/1399

**Comment by jwang47** [comment]
http://builds.cask.co/browse/CDAP-DUT874

File: cdap-cli/src/main/java/co/cask/cdap/cli/util/AsciiTable.java [hunk 1]
```diff
@@ -47,19 +47,35 @@
  */
 public class AsciiTable<T> {
 
+  private static final int DEFAULT_WIDTH = 80;
+  private static final int MIN_COLUMN_WIDTH = 5;
+  private static final Splitter NEWLINE_SPLITTER = Splitter.on(System.getProperty("line.separator"));
```
Comment (albertshau) [inline cdap-cli/src/main/java/co/cask/cdap/cli/util/AsciiTable.java]: seems like this should be passed in instead of read from system properties

**Comment by albertshau** [comment]
one small comment, otherwise lgtm

**Pull request merged**